src/core/src/Environment.C

```diff
@@ -1247,26 +1257,9 @@ FullEnvironment::construct (RawType_MPI_Comm inputComm,
   std::cout << "In FullEnv, finished dealing with MPI initially" << std::endl;
 #endif
 
-  //////////////////////////////////////////////////
   // Read options
-  //////////////////////////////////////////////////
-  // If NULL, we create one
-  if (m_optionsObj == NULL) {
-    // If there's an input file, we grab the options from there.  Otherwise the
-    // defaults are used
-    if (m_optionsInputFileName != "") {
-#ifndef QUESO_DISABLE_BOOST_PROGRAM_OPTIONS
-      m_allOptionsMap.reset(new boost::program_options::variables_map());
-      m_allOptionsDesc.reset(new boost::program_options::options_description("Allowed options"));
-#endif  // QUESO_DISABLE_BOOST_PROGRAM_OPTIONS
-
-      readOptionsInputFile();
-
-      m_input->parse_input_file(m_optionsInputFileName);
-    }
-
-    m_optionsObj.reset(new EnvOptionsValues(this, prefix));
-  }
+  if (!m_optionsInputFileName.empty())
+    readOptionsInputFile(prefix);
```

Comment:
This is a semantic change, not just an improved capability, so let's be wary of regressions.  I can't imagine anyone was actually *using* the behavior of "pass an input file name and a non-null option object pointer in and expect the former to be ignored", but I've seen stranger code in the past.